vmm/src/api/http.rs

```diff
@@ -68,72 +63,60 @@ lazy_static! {
     };
 }
 
-fn http_serve<T: Read + Write>(
-    http_connection: &mut HttpConnection<T>,
-    api_notifier: EventFd,
-    api_sender: Sender<ApiRequest>,
-) {
-    if http_connection.try_read().is_err() {
-        http_connection.enqueue_response(Response::new(
-            Version::Http11,
-            StatusCode::InternalServerError,
-        ));
-
-        return;
-    }
-
-    while let Some(request) = http_connection.pop_parsed_request() {
-        let sender = api_sender.clone();
-        let path = request.uri().get_abs_path().to_string();
-        let mut response = match HTTP_ROUTES.routes.get(&path) {
-            Some(route) => match api_notifier.try_clone() {
-                Ok(notifier) => route.handle_request(&request, notifier, sender),
-                Err(_) => Response::new(Version::Http11, StatusCode::InternalServerError),
-            },
-            None => Response::new(Version::Http11, StatusCode::NotFound),
-        };
+fn handle_http_request(
+    request: &Request,
+    api_notifier: &EventFd,
+    api_sender: &Sender<ApiRequest>,
+) -> Response {
+    let path = request.uri().get_abs_path().to_string();
+    let mut response = match HTTP_ROUTES.routes.get(&path) {
+        Some(route) => match api_notifier.try_clone() {
+            Ok(notifier) => route.handle_request(&request, notifier, api_sender.clone()),
+            Err(_) => Response::new(Version::Http11, StatusCode::InternalServerError),
+        },
+        None => Response::new(Version::Http11, StatusCode::NotFound),
+    };
 
-        response.set_server("Cloud Hypervisor API");
-        response.set_content_type(MediaType::ApplicationJson);
-        http_connection.enqueue_response(response);
-    }
+    response.set_server("Cloud Hypervisor API");
+    response.set_content_type(MediaType::ApplicationJson);
+    response
 }
 
 pub fn start_http_thread(
     path: &str,
     api_notifier: EventFd,
     api_sender: Sender<ApiRequest>,
 ) -> Result<thread::JoinHandle<Result<()>>> {
-    let listener = UnixListener::bind(path).map_err(Error::Bind)?;
-    let pool = ThreadPool::new(NUM_THREADS);
+    std::fs::remove_file(path).unwrap_or_default();
```

Comment:
Are you doing `.unwrap_or_default()` to tolerate the error i.e. the file doesn't exist?

Reply:
Yes. It's a convenience, we don't want to error out if the file removal could not happen when e.g. the file does not exist.